lib/pure/md5.nim

```diff
@@ -97,10 +98,62 @@ proc decode(dest: var openArray[uint8], src: openArray[uint32]) =
     dest[i+3] = uint8(src[j] shr 24 and 0xff'u32)
     inc(i, 4)
 
-proc transform(buffer: pointer, state: var MD5State) =
+const openArrayTest = compiles do:
```

Comment:
I really hate how these end up looking. Don't know if people think it's worth it.

Reply:
It's not good, we need to do something. The compiler could use its own version of md5.nim so that bootstrapping keeps working (the compiler doesn't need md5.nim at its own compile-time either) and then md5.nim can be written in the one good way that is supported with Nim devel / 1.6.